src/Components/User/Saga.js

```diff
@@ -0,0 +1,28 @@
+import { call, put, takeLatest, take } from 'redux-saga/effects'
+import Api from '../../utils/api';
+
+function* fetchAllUsers() {
+   try {
+    const users = yield call(Api.user.getAllUsers);
+    yield put({ type: "SET_USERS", users });
+   } catch (e) {
+       // catch e.message
+   }
+}
+
+function* fetchUser(username, password) {
+    try {
+     const user = yield call(Api.user.getUser(username, password));
+     yield put({ type: "SET_USER", user });
+    } catch (e) {
+        // catch e.message
```

Comment:
one suggestion here, **not mandatory**

can we also manage the error condition as well here? like

`yield put({ type: "SET_ERR", err: 'Fetch failed' });`

in reducer 

`case 'SET_ERR':
        return {
          err: action.err
        }`

in sleector 

`const getError = () =>
  createSelector(
    UserReducerState,
    simpleState => simpleState.err,
  );`

then we can show the corresponding error in our UI